server/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -600,4 +572,49 @@ protected LogicalPlan visitAnalyzedInsertStatement(AnalyzedInsertStatement state
             );
         }
     }
+
+    private static final List<Rule<?>> ITERATIVE_OPTIMIZER_RULES = List.of(
+        new RemoveRedundantFetchOrEval(),
+        new MergeAggregateAndCollectToCount(),
+        new MergeAggregateRenameAndCollectToCount(),
+        new MergeFilters(),
+        new MoveFilterBeneathRename(),
+        new MoveFilterBeneathFetchOrEval(),
+        new MoveFilterBeneathOrder(),
+        new MoveFilterBeneathProjectSet(),
+        new MoveFilterBeneathJoin(),
+        new MoveFilterBeneathCorrelatedJoin(),
+        new MoveFilterBeneathUnion(),
+        new MoveFilterBeneathGroupBy(),
+        new MoveFilterBeneathWindowAgg(),
+        new MoveLimitBeneathRename(),
+        new MoveLimitBeneathEval(),
+        new MergeFilterAndCollect(),
+        new RewriteFilterOnOuterJoinToInnerJoin(),
+        new MoveOrderBeneathUnion(),
+        new MoveOrderBeneathNestedLoop(),
+        new MoveOrderBeneathFetchOrEval(),
+        new MoveOrderBeneathRename(),
+        new DeduplicateOrder(),
+        new OptimizeCollectWhereClauseAccess(),
+        new RewriteGroupByKeysLimitToLimitDistinct(),
+        new MoveConstantJoinConditionsBeneathNestedLoop(),
+        new RewriteNestedLoopJoinToHashJoin()
+    );
+
+    private static final List<Rule<?>> FETCH_OPTIMIZER_RULES = List.of(
+        new RemoveRedundantFetchOrEval(),
+        new RewriteToQueryThenFetch()
+    );
+
+    private static final List<Rule<?>> WRITE_OPTIMIZER_RULES =
+        List.of(new RewriteInsertFromSubQueryToInsertFromValues());
```

Comment:
I'd prefer declaring static definitions on top of the class, especially as these are used inside the ctor.